packages/core/src/common/registry/resolve-id-registry.ts

```diff
@@ -0,0 +1,56 @@
+import Factory from './factory';
+import Registry from './registry';
+
+export default class ResolveIdRegistry<TType, TToken extends { [key: string]: unknown }> {
+    private _registry: Registry<TType>;
+
+    constructor() {
+        this._registry = new Registry({
+            tokenResolver: this._resolveToken.bind(this),
+            useFallback: false,
+        });
+    }
+
+    get(resolveId: TToken): TType {
+        return this._registry.get(this._encodeToken(resolveId));
+    }
+
+    register(resolveId: TToken, factory: Factory<TType>): void {
+        this._registry.register(this._encodeToken(resolveId), factory);
+    }
+
+    private _encodeToken(resolveId: TToken): string {
+        return btoa(JSON.stringify(resolveId));
+    }
+
+    private _decodeToken(token: string): TToken {
+        return JSON.parse(atob(token));
+    }
+
+    private _resolveToken(token: string, registeredTokens: string[]): string | undefined {
+        const query = this._decodeToken(token);
+        const results: Array<{ token: string; matches: number }> = [];
+
+        for (const [, registeredToken] of Object.entries(registeredTokens)) {
```

Comment:
Just out of curiosity is there a reason why we use Object.entries for `registeredTokens` instead of a simple `registeredTokens.forEach` ?

Reply:
Ah I didn't realise `registeredTokens` is `string[]` so a simple `forEach` should do.